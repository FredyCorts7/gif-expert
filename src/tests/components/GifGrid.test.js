import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('Testing to <GifGrid/>', () => {
  const category = 'Demon Slayer';

  test('should be the same component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when load images', () => {
    const gifs = [
      {
        id: '1f23yj2',
        title: 'Somethings',
        url: 'https://somethings.png'
      },
      {
        id: '758dg5sa',
        title: 'Anymore',
        url: 'https://anymore.png'
      }
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
