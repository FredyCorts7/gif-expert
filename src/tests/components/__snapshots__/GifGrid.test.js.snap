// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <GifGrid/> should be the same component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Demon Slayer
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing to <GifGrid/> should show items when load images 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Demon Slayer
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1f23yj2"
      key="1f23yj2"
      title="Somethings"
      url="https://somethings.png"
    />
    <GifGridItem
      id="758dg5sa"
      key="758dg5sa"
      title="Anymore"
      url="https://anymore.png"
    />
  </div>
</Fragment>
`;
